src/__tests__/useMap.test.ts

```diff
@@ -0,0 +1,115 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import useMap from '../useMap';
+
+const setUp = (initialMap?: object) => renderHook(() => useMap(initialMap));
+
+it('should init map and utils', () => {
+  const { result } = setUp({ foo: 'bar', a: 1 });
+  const [map, utils] = result.current;
+
+  expect(map).toEqual({ foo: 'bar', a: 1 });
+  expect(utils).toStrictEqual({
+    get: expect.any(Function),
+    set: expect.any(Function),
+    remove: expect.any(Function),
+    reset: expect.any(Function),
+  });
+});
+
+it('should init empty map if not initial object provided', () => {
+  const { result } = setUp();
+
+  expect(result.current[0]).toEqual({});
+});
+
+it('should get corresponding value for existing provided key', () => {
+  const { result } = setUp({ foo: 'bar', a: 1 });
+  const [, utils] = result.current;
+
+  let value;
+  act(() => {
+    // @ts-ignore
+    value = utils.get('a');
```

Comment:
I'm getting a TS error that I don't even understand so I just flagged it to be ignored.

Reply:
If you replace

```ts
const { result } = setUp({ foo: 'bar', a: 1 });
```

by

```ts
const { result } = renderHook(() => useMap({ foo: 'bar', a: 1 }));
```

the error will disappear.

But if you will add `<object>` like

```ts
const { result } = renderHook(() => useMap<object>({ foo: 'bar', a: 1 }));
```

the error will appear again.

It happens because `useMap` hook things that the object is of the type `object` but should be of the type `{foo: string, a: number}`.

---

Alternatively, to fix it you could do

```ts
const setUp = <T extends object>(initialMap?: T) => renderHook(() => useMap(initialMap));

// ...

const { result } = setUp<{ foo: string, a: number }>({ foo: 'bar', a: 1 });
```

Reply:
Cool! I'll fix the ts-ignore flags on my next PR. Improving my TS skills was another motivation to contribute to this repo :)